File: splearn/src/test/java/com/splearn/domain/MemberFixture.java
package com.splearn.domain;

public class MemberFixture {
    public static MemberRegisterRequest createMemberRegisterRequest() {
        return new MemberRegisterRequest("cham@app.com", "cham", "secret");
    }

    public static PasswordEncoder createPasswordEncoder() {
        return new PasswordEncoder() {
            @Override
            public String encode(String password) {
                return password.toUpperCase();
            }

            @Override
            public boolean matches(String password, String passwordHash) {
                return encode(password).equals(passwordHash);
            }
        };
    }

    public static MemberRegisterRequest createMemberRegisterRequest(String email) {
        return new MemberRegisterRequest(email, "cham", "secret");
    }
}

File: splearn/src/test/kotlin/cham/splearn/domain/MemberFixture.kt
package cham.splearn.domain

fun createMemberRegisterRequest(email: String): MemberRegisterRequest = MemberRegisterRequest(email, "cham123", "secret1234")

fun createMemberRegisterRequest(): MemberRegisterRequest = MemberRegisterRequest("cham@splearn.com", "cham123", "secret1234")


fun createPasswordEncoder(): PasswordEncoder = object : PasswordEncoder {
    override fun encode(password: String): String {
        return password.reversed()
    }

    override fun matches(password: String, passwordHash: String): Boolean {
        return encode(password) == passwordHash
    }
}